CombinatorialGames/Game/IGame.lean

```diff
@@ -1303,13 +1322,33 @@ theorem invRec (x : IGame) {P : ∀ y ∈ x⁻¹.leftMoves, Prop} {Q : ∀ y ∈
   · simpa
   all_goals simp only [Bool.false_eq_true]
   on_goal 1 => apply left₁
-  on_goal 4 => apply left₂
-  on_goal 8 => apply right₁
-  on_goal 12 => apply right₂
-  any_goals simpa [inv_eq, InvTy.val]
+  on_goal 5 => apply left₂
+  on_goal 9 => apply right₁
+  on_goal 13 => apply right₂
+  any_goals simpa [inv_eq hx, InvTy.val]
   all_goals first |
-    exact ((equivShrink {y ∈ x.leftMoves | 0 < y}).symm _).2.1 |
-    exact ((equivShrink {y ∈ x.leftMoves | 0 < y}).symm _).2.2
+    exact ((equivShrink {y ∈ _ | 0 < y}).symm _).2.1 |
+    exact ((equivShrink {y ∈ _ | 0 < y}).symm _).2.2
+
+/-- An induction principle on left and right moves of `x⁻¹`. -/
+theorem invRec {x : IGame} (hx : 0 < x)
+    {P : ∀ y ∈ x⁻¹.leftMoves, Prop} {Q : ∀ y ∈ x⁻¹.rightMoves, Prop}
+    (zero : P 0 (zero_mem_leftMoves_inv hx))
+    (left₁ : ∀ y (hy : 0 < y) (hyx : y ∈ x.rightMoves), ∀ a (ha : a ∈ x⁻¹.leftMoves), P a ha →
+      P _ (invOption_right_left_mem_leftMoves_inv hx hy hyx ha))
+    (left₂ : ∀ y (hy : 0 < y) (hyx : y ∈ x.leftMoves), ∀ a (ha : a ∈ x⁻¹.rightMoves), Q a ha →
+      P _ (invOption_left_right_mem_leftMoves_inv hx hy hyx ha))
+    (right₁ : ∀ y (hy : 0 < y) (hyx : y ∈ x.leftMoves), ∀ a (ha : a ∈ x⁻¹.leftMoves), P a ha →
+      Q _ (invOption_left_left_mem_rightMoves_inv hx hy hyx ha))
+    (right₂ : ∀ y (hy : 0 < y) (hyx : y ∈ x.rightMoves), ∀ a (ha : a ∈ x⁻¹.rightMoves), Q a ha →
+      Q _ (invOption_right_right_mem_rightMoves_inv hx hy hyx ha)) :
+    (∀ y (hy : y ∈ x⁻¹.leftMoves), P y hy) ∧ (∀ y (hy : y ∈ x⁻¹.rightMoves), Q y hy) := by
+  apply invRec' hx zero
+  on_goal 1 => convert left₁ using 6 with _ ha
+  on_goal 2 => convert left₂ using 6 with _ ha
+  on_goal 3 => convert right₁ using 6 with _ ha
+  on_goal 4 => convert right₂ using 6 with _ ha
+  all_goals simp_rw [invOption_eq ha]
```

Comment:
(Not any review comment, but I must ask: is there any meaningful difference between `simp_rw [h1]` and `simp only [h1]` past `simp_rw`'s ease of extension?)

Reply:
I think they're the same, `simp_rw [h₁, h₂, h₃]` basically amounts to `simp only [h₁]; simp only [h₂]; simp only [h₃]`.